enhancements/machine-api/aws-efa-networking.md

```diff
@@ -0,0 +1,253 @@
+)
+```
+
+### Implementation Details/Notes/Constraints
+
+To enable Machine API to create EFA enabled instances, the above API field will need to be added to the AWS Machine
+ProviderConfig. Once that field is added, if set by a user, we will set the interface to the correct value while
+creating new instances.
+
+```go
+  if machineProviderConfig.NetworkInterfaceType != "" {
+		networkInterfaces[0].InterfaceType = aws.String(string(machineProviderConfig.NetworkInterfaceType))
+	}
+```
+
+It is important to note that OpenShift only supports a single network interface on instances created by Machine API
+today, as such, the EFA network interface will be the primary network interface for the instance.
+
+#### What is EFA?
+
+To quote the [AWS documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html):
+
+> An Elastic Fabric Adapter (EFA) is a network device that you can attach to your Amazon EC2 instance to accelerate
+  High Performance Computing (HPC) and machine learning applications. EFA enables you to achieve the application
+  performance of an on-premises HPC cluster, with the scalability, flexibility, and elasticity provided by the AWS
+  Cloud.
+>
+> EFA provides lower and more consistent latency and higher throughput than the TCP transport traditionally used in
+  cloud-based HPC systems. It enhances the performance of inter-instance communication that is critical for scaling HPC
+  and machine learning applications. It is optimized to work on the existing AWS network infrastructure and it can
+  scale depending on application requirements.
+
+### Risks and Mitigations
+
+#### This feature cannot be leveraged for control plane instances
+
+Due to limitations of the libraries that the installer uses to create control plane machines, there is currently no
+way in which we can enable the control plane machines to be created day 0 with the EFA network interface type.
+
+As this feature is only intended to be used with specific high performance workloads, which should not be running on
+control plane hosts, we do not expect end users to require EFA network interfaces for their control plane machines.
+
+#### Users will have to configure their own security groups
+
+To allow the feature to work, the security groups that contain the Machines must be configured to allow all traffic
+between the hosts within the security group. This is not configured by default on an IPI cluster.
+Users must manually make this change via the AWS console/cli.
+This prerequisite should be documented as a manual step to enable the feature.
+
+Additionally, as we have future plans to manage security groups for IPI clusters, these manual changes to the security
+groups may be difficult to adopt into the future management system. However, there is nothing to stop users of IPI
+clusters making changes such as this already, so this kind of change must be considered for any adoption plans already.
```

Comment:
> The other thing I was concerned about is that customers may try to automate the security-group-modifying step, in which case they would have to make assumptions about the existing OCP security groups (eg, that they have certain names or labels), and those assumptions may be broken in the future, breaking their automation.

This seems like a valid concern. Explicitly stating that we don't guarantee the shape of our security-group over time meaning that this can break is worthwhile.